### PR TITLE
[HCICD-464]: Release Notes Automation

### PR DESCRIPTION
**Pull request by mayur-hitachivantara**
This PR introduces two workflows:
- Release Notes Draft: Auto-generates and keeps release notes up-to-date upon PR merges into master.
- Update Release Documentation: Automatically updates release.md, syncs to Confluence, and sends Slack notifications for seamless collaboration.

These workflows streamline the release process and keep all stakeholders informed.

Credentials required to run these workflows

- Confluence User Email ID
- Confluence API Token
- Github Token 
- Slack Channel Name

### CODE REVIEW

File: .github/config.yml
Comment (cardosov) [inline .github/config.yml:1]: Maybe this file should have a more specific name. `config.yaml` seems to generic to just be inside the `.github` folder

**Review by cardosov — CHANGES_REQUESTED** [review]

File: .github/workflows/update-release-notes.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  release:
+    types: [released]
+
+env: 
+  file_path_md: ./docs/release.md                                # path to .md file
+  confluence_page_id: '30835966301'                              # The confluence page id where to write the output
+  sub_domain_name: 'hv-eng'                                      # confluence domain
+  confluence_page_title: 'Github Composite Actions Releases'  # Title of confluence page
+  slack_channel_name: yukon-test                                 # slack channel name
+  user_for_confluence: mayur.kadam@hitachivantara.com            # user for API token of Confluence
```
[the remark this answers is not in the document]
Comment (mgaikwad2) [reply .github/workflows/update-release-notes.yml:12]: In addition to user change as mentioned by Vitor ,  please verify the target space where this page is being created ?

File: .github/config.yml
Comment (mayur-hitachivantara) [reply .github/config.yml:1]: Hii @cardosov  
Can we name the` config.yml `as `draft-config.yml `  or any suggestions for names?

File: .github/workflows/update-release-notes.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (mayur-hitachivantara) [reply .github/workflows/update-release-notes.yml:12]: yes, we need a service account for both Confluence and Github.

File: docs/release.md
[the remark this answers is not in the document]
Comment (mayur-hitachivantara) [reply docs/release.md:1]: Yes `docs/release.md` needs to exist beforehand

File: .github/config.yml
Comment (cardosov) [reply .github/config.yml:1]: `draft-release-config.yml`

File: .github/draft-release-config.yml
Comment (cardosov) [inline .github/draft-release-config.yml:1]: I know I've suggested the name to be `draft-release-config` but I've just noticed that this file is to be used by the `release-drafter` action, so, I think the name should be inline with that. `release-drafter-config.yml` seems a better fit.

**Review by cardosov — CHANGES_REQUESTED** [review]

File: .github/draft-release-config.yml
Comment (mayur-hitachivantara) [reply .github/draft-release-config.yml:1]: Ok Vitor I will rename the file

**Review by cardosov — APPROVED** [review]

File: .github/workflows/update-release-notes.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (cardosov) [reply .github/workflows/update-release-notes.yml:12]: I've approved but forgot about this service account thing

**Review by cardosov — CHANGES_REQUESTED** [review]

File: .github/workflows/update-release-notes.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (mayur-hitachivantara) [reply .github/workflows/update-release-notes.yml:12]: Yes, I will Check with Steve for the above service account for Jira and Github.

[the remark this answers is not in the document]
Comment (mayur-hitachivantara) [reply .github/workflows/update-release-notes.yml:12]: @cardosov Also can we add Releases Info Done until now for gh-composite-action in ` ./docs/release.md ` file.

[the remark this answers is not in the document]
Comment (cardosov) [reply .github/workflows/update-release-notes.yml:12]: Sure

[the remark this answers is not in the document]
Comment (cardosov) [reply .github/workflows/update-release-notes.yml:12]: Still not using the service account.

**Review by cardosov — CHANGES_REQUESTED** [review]

File: .github/workflows/update-release-notes.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (mayur-hitachivantara) [reply .github/workflows/update-release-notes.yml:12]: Waiting for steve reply for service accounts

File: .github/workflows/update-release-notes.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+          ref: ${{ github.event.release.target_commitish }}
+      
+      - name: Update Confluence Page
+        uses: cupcakearmy/confluence-markdown-sync@v1
+        with:
+          from: ${{ env.file_path_md }}
+          to: ${{ env.confluence_page_id }} 
+          cloud: ${{ env.sub_domain_name }}
+          user: ${{ env.user_for_confluence }}
+          token: ${{ secrets.HNC_JIRA_CONFLUENCE_TOKEN }}
```
Comment (mayur-hitachivantara) [inline .github/workflows/update-release-notes.yml:55]: Token Updated

File: .github/workflows/release-drafter.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      pull-requests: write
+    runs-on: [k8s]
+    steps:
+      # Drafts your next Release notes as Pull Requests are merged into "master"
+      - uses: release-drafter/release-drafter@v5
+        with:
+          config-name: release-drafter-config.yml
+          disable-autolabeler: false
+        env:
+          GITHUB_TOKEN: ${{ secrets.HV_ADMIN_GITHUB_TOKEN }}
```
Comment (mayur-hitachivantara) [inline .github/workflows/release-drafter.yml:31]: Token Updated

**Review by cardosov — APPROVED** [review]

**Review by vganpisetty — APPROVED** [review]

**Pull request merged**